File: 2 term/PB/PB Lab2.5/PB Lab2.5 py/Prism.py
from abc import ABC, abstractmethod


class Prism(ABC):
    def __init__(self, height, side_length):
        self.height = height
        self.side_length = side_length

    @abstractmethod
    def calculate_surface(self):
        pass

    @abstractmethod
    def calculate_base_surface(self):
        pass

    def calculate_volume(self):
        return self.calculate_base_surface() * self.height

File: 2 term/PB/PB Lab2.5/PB Lab2.5 cs/Prism.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PB_Lab2._5_cs
{
    public abstract class Prism
    {
        public double Height { get; set; }
        public double SideLength { get; set; }
        public abstract double CalculateSurface();
        public abstract double CalculateBaseSurface();

        public double CalculateVolume()
        {
            return CalculateBaseSurface() * Height;
        }

        protected Prism(double height, double sideLength)
        {
            Height = height;
            SideLength = sideLength;
        }
    }
}
